src/main/java/com/google/sps/data/Meal.java

```diff
@@ -50,4 +51,28 @@ public String getDescription() {
     public String getType() {
         return this.type;
     }
+
+    /**
+     * Counts the number of keywords that are matching in recipe.
+     * @param params list of strings with keywords, that describe search request.
+     * @return amount of occurrences of keywords from params in this meal.
+     */
+    public int getFrequency(List<String> params) {
+        int frequency = 0;
+        for (String param : params) {
+            if (title.contains(param) ||
+                description.contains(param) ||
+                type.contains(param)) {
+                frequency++;
+                continue;
+            } 
+            for (String ingredient : ingredients) {
+                if (ingredient.contains(param)) {
+                    frequency++;
+                    continue;
```

Comment:
This `continue` will skip the remaining part of the inner loop and go to the next iteration. However, there's nothing happening *after* the `continue`, therefore this keyword does nothing (FWIU). This `continue` doesn't really do anything.
As a result, if a meal has 3 ingredients:
- chocolate flakes
- chocolate milk
- chocolate whatever

then as long as the word "chocolate" is not in the title / description / type, returned frequency will be 3. However, if the title *does* include the word "chocolate", we'll return early with `frequency` == 1. It looks like a tiny bug.

I believe you may have meant to break the outer loop instead. I think the simplest solution is to extract the second for loop to a method, e.g. `bool doesKeywordAppearInIngredients(String keyword)`. Then you could drop all those `continue` keywords from this method and just have a single `if`.